enhancements/installer/crc.md

```diff
@@ -0,0 +1,93 @@
+---
+title: code-ready-containers
+authors:
+reviewers:
+approvers:
+creation-date: 2020-03-02
+last-updated: 2020-03-02
+status: implemented
+see-also:
+  - "/enhancements/this-other-neat-thing.md"  
+replaces:
+superseded-by:
+---
+
+# Code Ready Containers (CRC)
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+This is where to call out areas of the design that require closure before deciding
+to implement the design.  For instance, 
+ > 1. This requires exposing previously private resources which contain sensitive
+  information.  Can we do this? 
+
+## Summary
+
+CRC functions by doing roughly....
+ 1.  Running a libvirt install, which is for ["development only"](https://github.com/openshift/installer/blob/ffc34e32fe4a71560f59312384daa87b401d6ec9/README.md) meaning OpenShift development only.
+ 2.  On a single node cluster, which is below the minimum of [three that are required](https://github.com/openshift/installer/blob/c904277e59dd947a8884265b2511034b05c38644/upi/openstack/inventory.yaml#L35) for a functional cluster.
+ 3.  Disable the CVO, which prevents all reconciliation of operators, top level status, and upgrades.
+ 4.  Deletes the monitoring stack, but not all monitor resources.
+     Removes the aggregated apiserver as well.
+ 5.  Scale insights to zero.
+ 6.  Force the router down to a single node.  Preventing smooth upgrades.
+ 7.  Twiddle the registry operator.
+     Changes for routes and PVs.
+ 8.  Disable the cloud credential operator.  Preventing reporting of failures to create and maintain creds for other operators.
+ 10. All other operators and operands are expected to function properly on a single node libvirt installation.
+ 
+Every one of these step must be considered supported when run by CRC in order to consider CRC a product.
+Every single one of these steps is, at the time of merging this enhancement, considered unsupported by the teams that own the components. 
```

Comment:
I'd also like a description that explains that end users don't run this installer. End users don't deal with libvirt. That this is all an internal only exercise and what end users get is an ISO that they can run in their own virtualization platform.

Reply:
> what end users get is a snapshot ISO that they can run in their own virtualization platform.

End users get a `crc` binary for their OS (windows/linux/macos), which embeds a so-called 'bundle' (a tarball with a VM disk image). The `crc` binary is able to create a VM using the native OS virtualization platform (hyperv/libvirt/hyperkit). We don't ship an ISO, and the end user does not get to pick the virtualization platform they want to use

Reply:
This also means, the installer is run as libvirt, and the resulting snapshot diskimage gets converted to supported disk images for the platforms respective hypervisor. At this moment, only Hyper-V needs a converted disk image (which also gets additional packages installed to communicate with the VMM, eg for IP address assignment, etc).